src/main/java/domain/pages/HomePage.java

```diff
@@ -10,46 +12,53 @@
     Boolean result;
     String regex = "^[0-9]{5}(?:-[0-9]{4})?$";
 
-    @FindBy (xpath = "(//a[@href = '/representatives'])[1]")
+    @FindBy (xpath = "//a[@title='Home']")
+    private WebElement homePageLogo;
+
+    @FindBy(xpath = "(//a[@href = '/representatives'])[1]")
     private WebElement representativesLink;
 
-    @FindBy (xpath = "//button[@value='Look up'][1]")
+    @FindBy(xpath = "//button[@value='Look up'][1]")
     private WebElement lookUpZipCodeButton;
 
-    @FindBy (xpath = "//input[@id='header-find-rep-zip-code']")
+    @FindBy(xpath = "//input[@id='header-find-rep-zip-code']")
     private WebElement inputZipCode;
 
-    @FindBy (xpath = "//*[contains(text(), 'You submitted an invalid Zip Code')] ")
+    @FindBy(xpath = "//*[contains(text(), 'You submitted an invalid Zip Code')]")
     private WebElement invalidZipCodeText;
 
-    @FindBy (xpath = "//img[@class='repPhoto'] ")
-    private WebElement representativePhoto;
+    @FindBy(xpath = "//img[@class='repPhoto']")
+    private WebElement imageRepresentative;
 
-    @FindBy (xpath = "//div[@name='RepeatReps']")
+    @FindBy(xpath = "(//img[@class='repPhoto'])[2]")
     private WebElement atLeastTwoRepresentativesFound;
 
-    @FindBy (xpath = "//input[@value='FIND YOUR REP BY ADDRESS']")
+    @FindBy(xpath = "//input[@value='FIND YOUR REP BY ADDRESS']")
     private WebElement findYourRepByAddressButton;
 
-    public void clickRepresentativesLink(){
+    public void clickRepresentativesLink() {
         representativesLink.click();
     }
 
-    public String enterZipCode(String zipCode) {
-        Pattern pattern =Pattern.compile(regex);
-        Matcher matcher =pattern.matcher(zipCode);
+    public void enterZipCodeAndSubmit(String zipCode) {
+        inputZipCode.clear();
         inputZipCode.sendKeys(zipCode);
         lookUpZipCodeButton.click();
-        if (matcher.matches()){
-            if(atLeastTwoRepresentativesFound.isDisplayed()) {
-                result = findYourRepByAddressButton.isDisplayed();
-            } else if(representativePhoto.isDisplayed())
-            result = representativePhoto.isDisplayed();
-        }
-        else {
-            result = invalidZipCodeText.isDisplayed();
-        }
-        return result.toString();
     }
 
+    public boolean isRepPhotoDisplayed(){
+        return imageRepresentative.isDisplayed();
+    }
+
+    public boolean IsAtleastTwoRepresentativesFound() {
+        return atLeastTwoRepresentativesFound.isDisplayed();
+    }
+
+    public boolean isFindYourRepButtonPresent(){
+         return findYourRepByAddressButton.isDisplayed();
+    }
+
+    public boolean isHomePageLogoDisplayed(){
```

Comment:
what does this return if HomePageLogo is not present?

Reply:
false

Reply:
sure?

Reply:
OK this is your challenge. In the above tests you are always testing the presence of a text/img etc. And during test you always see them. What you want verify that an error message is not present... Make up a scenario in this SUT and test this scenario.